docs/configuration.asciidoc

```diff
@@ -0,0 +1,252 @@
+
+`MaxDeadTimeout`::
+
+The maximum amount of time a node is allowed to marked dead.
+
+`MaximumRetries`::
+
+When a retryable exception occurs or status code is returned this controls the 
+maximum amount of times we should retry the call to {es}.
+
+`MaxRetryTimeout`::
+
+Limits the total runtime including retries separately from `RequestTimeout`. 
+When not specified defaults to `RequestTimeout` which itself defaults to 60 
+seconds.
+
+`MemoryStreamFactory`::
+
+Provides a memory stream factory.
+
+`NodePredicate`::
+
+Register a predicate to select which nodes that you want to execute API calls 
+on. Note that sniffing requests omit this predicate and always execute on all 
+nodes. When using an `IConnectionPool` implementation that supports reseeding of 
+nodes, this will default to omitting master only node from regular API calls. 
+When using static or single node connection pooling it is assumed the list of 
+node you instantiate the client with should be taken verbatim.
+
+`OnRequestCompleted`::
+
+Allows you to register a callback every time a an API call is returned.
+
+`OnRequestDataCreated`::
+
+An action to run when the `RequestData` for a request has been created.
+
+`PingTimeout`::
+
+The timeout in milliseconds to use for ping requests, which are issued to 
+determine whether a node is alive.
+
+`PrettyJson`::
+
+Provide hints to serializer and products to produce pretty, non minified json.
++
+Note: this is not a guarantee you will always get prettified json.
+
+`Proxy`::
+
+If your connection has to go through proxy, use this method to specify the 
+proxy url.
+
+`RequestTimeout`::
+
+The timeout in milliseconds for each request to {es}.
+
+`ServerCertificateValidationCallback`::
+
+Register a `ServerCertificateValidationCallback` per request.
+
+`SkipDeserializationForStatusCodes`::
+
+Configure the client to skip deserialization of certain status codes, for 
+example, you run {es} behind a proxy that returns an unexpected json format.
+
+`SniffLifeSpan`::
+
+Force a new sniff for the cluster when the cluster state information is older 
+than the specified timespan.
+
+`SniffOnConnectionFault`::
+
+Force a new sniff for the cluster state every time a connection dies.
+
+`SniffOnStartup`::
+
+Sniff the cluster state immediately on startup.
+
+`ThrowExceptions`::
+
+Instead of following a c/go like error checking on response. `IsValid` do throw 
+an exception (except when `SuccessOrKnownError` is false) on the client when a 
+call resulted in an exception on either the client or the {es} server.
++
+Reasons for such exceptions could be search parser errors, index missing 
+exceptions, and so on.
+
+`TransferEncodingChunked`::
+
+Whether the request should be sent with chunked Transfer-Encoding.
+
+`UserAgent`::
+
+The user agent string to send with requests. Useful for debugging purposes to 
+understand client and framework versions that initiate requests to {es}.
+
+
+==== Options on ConnectionSettings
```

Comment:
All options are now on the `ElasticsearchClientSettings` so these can be combined into the above list.

Reply:
I combined the two lists of settings into one via [2c434a9](https://github.com/elastic/elasticsearch-net/pull/6387/commits/2c434a9b52306b04cbc298496a6c858321b6f3a8).